src/vis/vis.js

```diff
@@ -136,18 +134,18 @@ var VisModel = Backbone.Model.extend({
 
     var windshaftClient = new WindshaftClient(windshaftSettings);
 
-    // Create the public Analysis Factory
-    this.analysis = new AnalysisFactory({
+    // Create the public Analysis Service
+    this._analysisService = new AnalysisService({
+      vis: this,
       apiKey: this.get('apiKey'),
-      authToken: this.get('authToken'),
-      analysisCollection: this._analysisCollection,
-      vis: this
+      authToken: this.get('authToken')
     });
 
-    var layersFactory = new LayersFactory({
-      visModel: this,
-      windshaftSettings: windshaftSettings
-    });
+    // Public wrapper exposing public methods.
+    this.analysis = {
+      analyse: this._analysisService.analyse.bind(this._analysisService),
```

Comment:
is the `.bind` required?

Reply:
Yes I think so because of the context this